lib/wit/gitrepo.py

```diff
@@ -89,6 +90,19 @@ def clone(self, source, name):
             else:
                 raise
 
+    # Use git clone's '--reference' to point at a local repository cache to copy objects/commits
+    # to save network traffic. Any missing objects/commits are downloaded from the true remote.
+    # Only newer git versions can use '--reference-if-able', so we emulate the 'if-able' bit.
```

Comment:
Python has first-class syntax support for documentation strings, which should be placed immediately below the function `def` line as a string (usually a multiline string with `"""`). We should follow the common Python convention, since it's pretty ubiquitous and tools for generating HTML docs from code and even IDEs will read from it.

See https://www.python.org/dev/peps/pep-0257/ for the spec and for a few examples.